aptos-move/framework/aptos-framework/sources/permissioned_signer.move

```diff
@@ -0,0 +1,620 @@
+///
+/// A permissioned signer is a restricted version of a signer. Functions `move_to` and
+/// `address_of` behave the same, and can be passed wherever signer is needed. However,
+/// code can internally query for the permissions to assert additional restrictions on
+/// the use of the signer.
+///
+/// A client which is interested in restricting access granted via a signer can create a permissioned signer
+/// and pass on to other existing code without changes to existing APIs. Core functions in the framework, for
+/// example account functions, can then assert availability of permissions, effectively restricting
+/// existing code in a compatible way.
+///
+/// After introducing the core functionality, examples are provided for withdraw limit on accounts, and
+/// for blind signing.
+module aptos_framework::permissioned_signer {
+    use std::signer;
+    use std::error;
+    use std::vector;
+    use std::option::{Option, Self};
+    use aptos_std::copyable_any::{Self, Any};
+    use aptos_std::simple_map::{Self, SimpleMap};
+    use aptos_framework::create_signer::create_signer;
+    use aptos_framework::transaction_context::generate_auid_address;
+    use aptos_framework::timestamp;
+
+    friend aptos_framework::account;
+    friend aptos_framework::aptos_governance;
+    friend aptos_framework::code;
+    friend aptos_framework::coin;
+    friend aptos_framework::fungible_asset;
+    friend aptos_framework::object;
+    friend aptos_framework::stake;
+    friend aptos_framework::staking_proxy;
+    friend aptos_framework::voting;
+
+    #[test_only]
+    friend aptos_framework::permissioned_signer_tests;
+
+    /// Trying to grant permission using non-master signer.
+    const ENOT_MASTER_SIGNER: u64 = 1;
+
+    /// Cannot authorize a permission.
+    const ECANNOT_AUTHORIZE: u64 = 2;
+
+    /// Access permission information from a master signer.
+    const ENOT_PERMISSIONED_SIGNER: u64 = 3;
+
+    /// signer doesn't have enough capacity to extract permission.
+    const ECANNOT_EXTRACT_PERMISSION: u64 = 4;
+
+    /// permission handle has expired.
+    const E_PERMISSION_EXPIRED: u64 = 5;
+
+    /// storing extracted permission into a different signer.
+    const E_PERMISSION_MISMATCH: u64 = 6;
+
+    /// permission handle has been revoked by the original signer.
+    const E_PERMISSION_REVOKED: u64 = 7;
+
+    /// destroying permission handle that has already been revoked or not owned by the
+    /// given master signer.
+    const E_NOT_ACTIVE: u64 = 8;
+
+    const U256_MAX: u256 =
+        115792089237316195423570985008687907853269984665640564039457584007913129639935;
+
+    struct RevokePermissionHandlePermission has copy, store, drop {}
+
+    struct GrantedPermissionHandles has key {
+        active_handles: vector<address>
+    }
+
+    enum PermissionedHandle {
+        V1 {
+            master_account_addr: address,
+            permissions_storage_addr: address
```

Comment:
For example, for this field it would be good to know it is a generated address for which we can obtain a signer.